django/template/defaulttags.py

```diff
@@ -1564,3 +1567,186 @@ def do_with(parser, token):
     nodelist = parser.parse(("endwith",))
     parser.delete_first_token()
     return WithNode(None, None, nodelist, extra_context=extra_context)
+
+
+class TemplateProxy:
+    """
+    A lightweight Template lookalike used for template partials.
+
+    Wraps nodelist as partial, in order to bind context.
+    """
+
+    def __init__(self, nodelist, origin, name):
+        self.nodelist = nodelist
+        self.origin = origin
+        self.name = name
+
+    def get_exception_info(self, exception, token):
+        template = self.origin.loader.get_template(self.origin.template_name)
+        return template.get_exception_info(exception, token)
+
+    def find_partial_source(self, full_source, partial_name):
```

Comment:
This method may not be fully tested, could you please point me to the tests covering the non-trivial logic described here?

In particular I think we need a new test file covering all the corner cases in `syntax_tests/test_partials` or similar.